android/app/src/main/java/org/smartregister/fhircore/activity/RecordVaccineActivity.kt

```diff
@@ -62,9 +77,42 @@ class RecordVaccineActivity : AppCompatActivity() {
           QuestionnaireActivity.QUESTIONNAIRE_FRAGMENT_TAG
         ) as
           QuestionnaireFragment
-      val vaccineSelected = questionnaireFragment.getQuestionnaireResponse()
+      val questionnaireResponse = questionnaireFragment.getQuestionnaireResponse()
+
+      val iParser: IParser = FhirContext.forR4().newJsonParser()
+      val questionnaire =
+        iParser.parseResource(
+          org.hl7.fhir.r4.model.Questionnaire::class.java,
+          viewModel.questionnaire
+        ) as
+          Questionnaire
+
+      // TODO Replace manual mapping with resource mapper
```

Comment:
Looks like we have manual mapping from covaxx here. The dynamic implementation is in progress. @ekigamba could you advise?

Reply:
@syedowaisali90 I'm going to add a DNM on this PR for now

Reply:
how far out is that work? if it's not going to happen next week, I'd rather get this merged and create an issue for this todo. So that the rest of this PR doesn't go out of date